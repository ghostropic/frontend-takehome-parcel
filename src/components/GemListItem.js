import React from 'react'
import styled from 'styled-components'
import {
  Card,
  CardHeader,
  CardContent,
  Typography
} from '@material-ui/core'

import Button from './Button'

const StyledCard = styled(Card)`
position: relative;
top: 20%;
margin: 20px;
`

export const StyledButton = styled(Button)({
  background: 'linear-gradient(45deg, #FE6B8B 30%, #FF8E53 90%)',
  border: 0,
  borderRadius: 3,
  boxShadow: '0 3px 5px 2px rgba(255, 105, 135, .3)',
  color: 'white',
  height: 48,
  padding: '0 30px',
});

export default class GemListItem extends React.Component {
  constructor(props) {
    super(props)
  }

  render () {
    const {
      name,
      info, 
      version,
      authors,
      downloads,
      project_uri,
      handleSaveGem,
      handleRemoveGem,
      listType,
      isSaved
    } = this.props
    return (
      <StyledCard>
        <CardContent>
          <Typography component="h5" variant="h5">
          <p>{name}</p>
          </Typography>
          <p>{info}</p>
          <p>{`Version: ${version}`}</p>
          <p>{`Authors: ${authors}`}</p>
          <p>{`Downloads: ${downloads}`}</p>
          <p>{project_uri}</p>
          <StyledButton
            listType={listType}
            text={isSaved ? 'remove' : 'save'}
            onClick={(e) => {
              handleRemoveGem
                ? handleRemoveGem(this.props)
                : handleSaveGem(this.props)
            }}
            >
          </StyledButton>
        </CardContent>
      </StyledCard>
    )
  }
}