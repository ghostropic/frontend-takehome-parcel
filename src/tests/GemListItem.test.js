import React from 'react'
import { shallow } from 'enzyme'

import GemListItem from '../components/GemListItem';
import gems from './fixture'

let wrapper
const gem = gems[0]

beforeEach(() => {
  const handleRemoveGem = jest.fn()
  const handleSaveGem = jest.fn()
  wrapper = shallow(<GemListItem
    handleRemoveGem={handleRemoveGem}
    handleSaveGem={handleSaveGem}
    name={gem.name}
    info={gem.info}
    version={gem.version}
    author={gem.author}
    downloads={gem.downloads}
    project_uri={gem.project_uri}
     />)
})

test('should render GemListItem correctly', () => {
  expect(wrapper).toMatchSnapshot()
})