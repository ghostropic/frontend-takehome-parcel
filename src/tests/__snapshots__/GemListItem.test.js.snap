// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render GemListItem correctly 1`] = `
<Styled(WithStyles(Card))>
  <WithStyles(CardContent)>
    <WithStyles(Typography)
      component="h5"
      variant="h5"
    >
      <p>
        cake
      </p>
    </WithStyles(Typography)>
    <p>
      Save your fork, there's cake!
    </p>
    <p>
      Version: 0.6.3
    </p>
    <p>
      Authors: undefined
    </p>
    <p>
      Downloads: 150214
    </p>
    <p>
      https://rubygems.org/gems/cake
    </p>
    <Styled(Button)
      onClick={[Function]}
      text="save"
    />
  </WithStyles(CardContent)>
</Styled(WithStyles(Card))>
`;
